client/src/components/mergeReportsDialog/mergeReportsDialog.js

```diff
@@ -0,0 +1,204 @@
+          <ul>
+            {invalidScans.map(scan => {
+              return <li key={scan.id}>{scan.name}</li>;
+            })}
+          </ul>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderButtons() {
+    const { onClose } = this.props;
+    const validCount = _.size(this.getValidScans());
+
+    if (validCount === 0) {
+      return (
+        <Button bsStyle="primary" className="btn-cancel" onClick={onClose}>
+          Close
+        </Button>
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <Button bsStyle="default" className="btn-cancel" onClick={onClose}>
+          Cancel
+        </Button>
+        <Button bsStyle="primary" type="submit" onClick={this.mergeScanResults}>
+          Merge
+        </Button>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    const { show, scans, onClose } = this.props;
+
+    if (!scans || scans.length === 0 || !scans[0]) {
+      return null;
+    }
+    const validCount = _.size(this.getValidScans());
+    const invalidCount = _.size(this.getInvalidScans());
+
+    let icon;
+    let heading;
+    if (validCount < 2) {
+      icon = <Icon type="pf" name="error-circle-o" />;
+      heading = 'This action is invalid. You must select at least two scans with successful most recent scans.';
+    } else if (invalidCount > 0) {
+      icon = <Icon type="pf" name="warning-triangle-o" />;
+      heading = 'Warning, only the selected scans with successful most recent scans will be included in the report.';
+    } else {
+      icon = <Icon type="pf" name="info" />;
+    }
+
+    return (
+      <Modal show={show} onHide={onClose}>
+        <Modal.Header>
+          <button className="close" onClick={onClose} aria-hidden="true" aria-label="Close">
+            <Icon type="pf" name="close" />
+          </button>
+          <Modal.Title>Merge Reports</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="merge-reports-body">
+            <span className="merge-reports-icon">{icon}</span>
+            <span>
+              <div className="merge-reports-heading">
+                <span>{heading}</span>
+              </div>
+              <div>
+                {this.renderValidScans()}
+                {this.renderInvalidScans()}
+              </div>
+            </span>
+          </div>
+        </Modal.Body>
+        <Modal.Footer>{this.renderButtons()}</Modal.Footer>
+      </Modal>
+    );
+  }
+}
+
+MergeReportsDialog.propTypes = {
+  mergeScans: PropTypes.func,
+  getMergedScanReportDetailsCsv: PropTypes.func,
+  show: PropTypes.bool.isRequired,
+  scans: PropTypes.array,
+  onClose: PropTypes.func
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  mergeScans: data => dispatch(mergeScans(data)),
+  getMergedScanReportDetailsCsv: id => dispatch(getMergedScanReportDetailsCsv(id))
+});
+
+const mapStateToProps = function(state) {
+  return {};
+};
```

Comment:
we could probably just remove the `mapStateToProps`

Reply:
I could pass noop maybe?